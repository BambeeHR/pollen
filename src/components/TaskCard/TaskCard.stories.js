import TaskCard from './TaskCard.vue';

export default {
  title: 'TaskCard',
  component: TaskCard,
};

export const DocumentsAwaitingSignature = () => ({
  components: { TaskCard },
  data() {
    return {
      items: [
        {
          title: 'Discrimination & Harassment Policy',
          description: '4 Days Ago',
          actions: [
            {
              // These are all just props that will be spread onto a BaseButton
              // component in TaskCardItem, with the exception of the "handler"
              // function which is bound to the `click` event.
              disabled: true,
              flat: true,
              label: 'Signed',
              preIcon: 'check',
              variant: 'secondary',
            },
          ],
        },
        {
          title: 'Dress Code Policy',
          description: '4 Days Ago',
          actions: [
            {
              flat: true,
              label: 'Preview',
              handler: () => {
                console.log('clicked preview');
              },
            },
            {
              label: 'Sign',
              preIcon: 'pen',
              variant: 'inverted',
              handler: () => {
                console.log('clicked "Dress Code Policy"');
              },
            },
          ],
        },
        {
          title: 'Computer Use Policy',
          description: '4 Days Ago',
          actions: [
            {
              flat: true,
              label: 'Preview',
              handler: () => {
                console.log('clicked preview');
              },
            },
            {
              label: 'Sign',
              preIcon: 'pen',
              variant: 'inverted',
              handler: () => {
                console.log('clicked "Computer Use Policy"');
              },
            },
          ],
        },
        {
          title: 'Paid Time Off Policy',
          description: '4 Days Ago',
          actions: [
            {
              flat: true,
              label: 'Preview',
              handler: () => {
                console.log('clicked preview');
              },
            },
            {
              label: 'Sign',
              preIcon: 'pen',
              variant: 'inverted',
              handler: () => {
                console.log('clicked "Paid Time Off Policy"');
              },
            },
          ],
        },
        {
          title: 'I9 Form',
          description: 'Signed 4 Days Ago',
          attribution: {
            name: 'Sarah Golden',
            image: 'https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc',
          },
          actions: [
            {
              flat: true,
              label: 'Preview',
              handler: () => {
                console.log('clicked preview');
              },
            },
            {
              label: 'Counter Sign',
              preIcon: 'pen',
              variant: 'inverted',
              handler: () => {
                console.log('clicked "I9 Form"');
              },
            },
          ],
        },
      ],
      primaryAction: {
        label: 'Sign All',
        preIcon: 'pen',
        handler: () => {
          console.log('clicked sign all');
        },
      },
    };
  },
  template: `
    <div class="bg-gray-6 flex items-center justify-center w-full min-h-screen">
      <TaskCard
        title="Documents Awaiting Signature"
        :count="4"
        icon="signTablet"
        :items="items"
        :primary-action="primaryAction"
      />
    </div>
  `,
});

export const EmploymentNotices = () => ({
  components: { TaskCard },
  data() {
    return {
      items: [
        {
          title: 'Paid Family Leave Pamphlet DE 2511',
          description:
            "This pamphlet provides information on California's Paid Family Leave program",
          actions: [
            {
              flat: true,
              label: 'View',
              handler: () => {
                console.log('clicked view');
              },
            },
          ],
        },
        {
          title: "Worker's Compensation Pamphlet",
          description:
            "This pamphlet provides information on California's Worker's Compensation program.",
          actions: [
            {
              flat: true,
              label: 'View',
              handler: () => {
                console.log('clicked view');
              },
            },
          ],
        },
        {
          title: 'Sexual Harassment Pamphlet',
          description:
            'This pamphlet informs employees about the definition of Sexual Harassment and their rights in California.',
          actions: [
            {
              flat: true,
              label: 'View',
              handler: () => {
                console.log('clicked view');
              },
            },
          ],
        },
        {
          title: 'Victims of Domestic Violence Leave Notice',
          description:
            "This notice provides information about California's Victims of Domestic Violence Leave law.",
          actions: [
            {
              flat: true,
              label: 'View',
              handler: () => {
                console.log('clicked view');
              },
            },
          ],
        },
        {
          title: 'Disability Insurance Pamphlet - DE 2515',
          description:
            "This pamphlet provides information on California's Disability Insurance Programs.",
          actions: [
            {
              flat: true,
              label: 'View',
              handler: () => {
                console.log('clicked view');
              },
            },
          ],
        },
      ],
      primaryAction: {
        label: 'Acknowledge',
        handler: () => {
          console.log('clicked sign all');
        },
      },
    };
  },
  template: `
    <div class="bg-gray-6 flex items-center justify-center w-full min-h-screen">
      <TaskCard
        title="Employment Notices"
        description="Mandatory notices to employee"
        :count="5"
        icon="clipboard"
        :items="items"
        :primary-action="primaryAction"
      />
    </div>
  `,
});
