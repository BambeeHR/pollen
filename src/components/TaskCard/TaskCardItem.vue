<template>
  <div class="task-card-item">
    <div class="task-card-item__content">
      <div class="task-card-item__title">
        {{ title }}
      </div>
      <div v-if="description" class="task-card-item__description">
        {{ description }}
        <template v-if="attribution">
          By
          <UserAvatar
            v-bind="attribution"
            :size="$options.Sizes.TINY"
            class="task-card-item__avatar"
          />
        </template>
      </div>
    </div>
    <div v-if="actions" class="task-card-item__actions">
      <BaseButton
        v-for="{ handler, ...rest } in actions"
        :key="rest.label"
        v-bind="rest"
        class="task-card-item__action"
        :size="$options.Button.Sizes.SMALL"
        @click="() => (handler ? handler() : () => {})"
      />
    </div>
  </div>
</template>

<script>
import BaseButton from '../BaseButton/BaseButton.vue';
import Button from '../../constants/Button';
import UserAvatar, { Sizes } from '../UserAvatar/UserAvatar.vue';

export default {
  Button,
  Sizes,
  components: { BaseButton, UserAvatar },
  props: {
    actions: {
      type: Array,
      default: null,
    },
    attribution: {
      type: Object,
      default: null,
    },
    description: {
      type: String,
      default: null,
    },
    title: {
      type: String,
      required: true,
    },
  },
};
</script>

<style scoped>
.task-card-item {
  @apply flex flex-col mx-6 py-3;
}

.task-card-item__content {
  @apply flex-grow mb-4;
}

.task-card-item__title {
  @apply font-body-text font-medium text-gray-0;
}

.task-card-item__description {
  @apply font-body-small text-gray-3;
}

.task-card-item__avatar {
  @apply ml-2 align-middle;
}

.task-card-item__action {
  @apply ml-2;
}

.task-card-item__action.base-button--flat[disabled] {
  @apply opacity-100 px-0 text-secondary;
}

.task-card-item__action.base-button--flat[disabled] >>> .base-button__inner {
  @apply px-0;
}

.task-card-item__actions {
  @apply flex flex-shrink-0 items-center justify-end;
}

@media screen and (min-width: 768px) {
  .task-card-item {
    @apply flex-row;
  }

  .task-card-item__content {
    @apply mb-0;
  }
}
</style>
