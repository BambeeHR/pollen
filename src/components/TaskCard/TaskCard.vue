<template>
  <article class="task-card">
    <header class="task-card__header">
      <div v-if="icon || !!$slots['header-icon']" class="task-card__icon">
        <slot name="header-icon">
          <FancyIcon v-if="icon" :icon="icon" />
        </slot>
      </div>
      <div>
        <slot name="header-content">
          <component :is="titleTag" class="task-card__title">
            {{ title }}
            <CounterBadge
              v-if="count !== null"
              :value="count"
              class="align-middle"
            />
          </component>
          <div v-if="description" class="task-card__description">
            {{ description }}
          </div>
        </slot>
      </div>
    </header>
    <div
      v-if="!!$slots['primary-action'] || primaryAction"
      class="task-card__primary-action"
    >
      <slot name="primary-action">
        <BaseButton
          v-if="primaryAction"
          v-bind="primaryAction"
          type="button"
          @click="primaryAction.handler"
        />
      </slot>
    </div>
    <div v-if="!!$slots['item-list'] || items.length" class="task-card__items">
      <slot name="item-list">
        <TaskCardItem
          v-for="item in items"
          :key="item.id || item.title"
          v-bind="item"
        />
      </slot>
    </div>
  </article>
</template>

<script>
import BaseButton from '../BaseButton/BaseButton.vue';
import CounterBadge from '../CounterBadge/CounterBadge.vue';
import FancyIcon, { isValidIcon } from '../FancyIcon/FancyIcon.vue';
import TaskCardItem from './TaskCardItem.vue';

/**
 * import { TaskCard } from '@bambee/pollen';
 *
 * A card component for displaying a list of action items. This is designed to
 * largely be a layout component. Any heavy logic lifting should be done by a
 * wrapping component. See Storybook for examples.
 */
export default {
  components: { BaseButton, CounterBadge, FancyIcon, TaskCardItem },
  props: {
    /** The number of tasks within the card. */
    count: {
      type: Number,
      default: null,
    },
    /** A top-line description of the card's content. */
    description: {
      type: String,
      default: null,
    },
    /** An optional `FancyIcon` to display at the top of the card. */
    icon: {
      type: String,
      default: null,
      validator: (value) => !value || isValidIcon(value),
    },
    /**
     * An array of object items to display in the card. These items should have
     * a `title` string, a `description` string, an array of `actions` objects
     * that have a `label` string and `handler` function, and finally an
     * optional `attribution` object that has at least an `image` url or a
     * `name` string.
     */
    items: {
      type: Array,
      default: () => [],
    },
    /**
     * An object with at least a `handler` function. The object will be spread
     * on to a `BaseButton` component.
     */
    primaryAction: {
      type: Object,
      default: null,
    },
    /**
     * The top-line title to display.
     */
    title: {
      type: String,
      default: null,
    },
    /**
     * The HTML tag for the title of the task card, for better structuring a
     * page's semantics.
     */
    titleTag: {
      type: String,
      default: 'h2',
    },
  },
};
</script>

<style scoped>
.task-card {
  @apply bg-white flex flex-col flex-wrap max-w-5xl shadow-1 w-full;
}

.task-card__header {
  @apply flex items-center px-6 py-3;
}

.task-card__icon {
  @apply flex-shrink-0 mr-6 text-40;
}

.task-card__title {
  @apply font-heading-subtitle font-semibold;
}

.task-card__description {
  @apply font-body-small;
}

.task-card-item + .task-card-item {
  @apply border-gray-5 border-b-0 border-l-0 border-r-0 border-t border-solid;
}

.task-card__items {
  @apply border-b-0 border-gray-5 border-l-0 border-r-0 border-solid border-t py-4 w-full;
}

.task-card__primary-action {
  @apply flex-shrink-0 ml-auto mr-6 mb-6 order-last self-center;
}

.task-card__primary-action ~ .task-card__items {
  @apply border-b mb-4;
}

@media screen and (min-width: 768px) {
  .task-card {
    @apply flex-row rounded-md;
  }

  .task-card__header {
    @apply py-5;
  }

  .task-card__title {
    @apply font-heading-title m-0;
  }

  .task-card__description {
    @apply font-body-caption;
  }

  .task-card-item + .task-card-item {
    @apply border-0;
  }

  .task-card__primary-action {
    @apply mb-0 order-none;
  }

  .task-card__primary-action ~ .task-card__items {
    @apply border-b-0 mb-0;
  }
}
</style>
