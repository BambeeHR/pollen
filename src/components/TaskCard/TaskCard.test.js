import { mount } from '@vue/test-utils';
import TaskCard from './TaskCard.vue';
import {
  DocumentsAwaitingSignature,
  EmploymentNotices,
} from './TaskCard.stories';

describe('TaskCard', () => {
  test('renders default correctly', () => {
    const wrapper = mount(TaskCard);
    expect(wrapper.element).toMatchSnapshot();
  });
  test('renders signature example correctly', () => {
    const example = DocumentsAwaitingSignature();
    const wrapper = mount(example);
    expect(wrapper.element).toMatchSnapshot();
  });
  test('renders signature example correctly', () => {
    const example = EmploymentNotices();
    const wrapper = mount(example);
    expect(wrapper.element).toMatchSnapshot();
  });
});
