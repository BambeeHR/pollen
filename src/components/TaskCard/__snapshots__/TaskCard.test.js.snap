// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard renders default correctly 1`] = `
<article
  class="task-card"
>
  <header
    class="task-card__header"
  >
    <!---->
     
    <div>
      <h2
        class="task-card__title"
      >
        
          
          
        <!---->
      </h2>
       
      <!---->
    </div>
  </header>
   
  <!---->
   
  <div
    class="task-card__items"
  />
</article>
`;

exports[`TaskCard renders signature example correctly 1`] = `
<div
  class="bg-gray-6 flex items-center justify-center w-full min-h-screen"
>
  <article
    class="task-card"
  >
    <header
      class="task-card__header"
    >
      <div
        class="task-card__icon"
      >
        <svg
          class="fancy-icon"
        />
      </div>
       
      <div>
        <h2
          class="task-card__title"
        >
          
          Documents Awaiting Signature
          
          <div
            class="counter-badge align-middle counter-badge--medium counter-badge--primary"
          >
            
  4

          </div>
        </h2>
         
        <!---->
      </div>
    </header>
     
    <div
      class="task-card__primary-action"
    >
      <button
        class="base-button base-button--primary base-button--medium base-button--regular base-button--inline"
        handler="() => {
          console.log('clicked sign all');
        }"
        type="button"
      >
        <div
          class="base-button__inner"
        >
          <svg
            aria-hidden="true"
            class="base-icon svg-inline--fa fa-pen fa-w-16 base-button__icon base-button__icon--pre"
            data-icon="pen"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
              fill="currentColor"
            />
          </svg>
          Sign All
        </div>
      </button>
    </div>
     
    <div
      class="task-card__items"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </article>
</div>
`;

exports[`TaskCard renders signature example correctly 2`] = `
<div
  class="bg-gray-6 flex items-center justify-center w-full min-h-screen"
>
  <article
    class="task-card"
  >
    <header
      class="task-card__header"
    >
      <div
        class="task-card__icon"
      >
        <svg
          class="fancy-icon"
        />
      </div>
       
      <div>
        <h2
          class="task-card__title"
        >
          
          Employment Notices
          
          <div
            class="counter-badge align-middle counter-badge--medium counter-badge--primary"
          >
            
  5

          </div>
        </h2>
         
        <div
          class="task-card__description"
        >
          
          Mandatory notices to employee
        
        </div>
      </div>
    </header>
     
    <div
      class="task-card__primary-action"
    >
      <button
        class="base-button base-button--primary base-button--medium base-button--regular base-button--inline"
        handler="() => {
          console.log('clicked sign all');
        }"
        type="button"
      >
        <div
          class="base-button__inner"
        >
          Acknowledge
        </div>
      </button>
    </div>
     
    <div
      class="task-card__items"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </article>
</div>
`;
